app/Http/Controllers/Admin/ProviderController.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\ProviderRequest;
+use App\Http\Resources\ProviderResource;
+use App\Models\Provider;
+use Illuminate\Http\Response as InertiaResponse;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Storage;
+use Inertia\Inertia;
+use Inertia\Response;
+
+class ProviderController extends Controller
+{
+    public const ITEMS_PER_PAGE = 15;
+
+    public function index(): Response
+    {
+        $providers = Provider::query()
+            ->search("name")
+            ->orderByName()
+            ->orderByTimeRange()
+            ->paginate(self::ITEMS_PER_PAGE)
+            ->withQueryString();
+
+        return Inertia::render("Providers/Index", [
+            "providers" => ProviderResource::collection($providers),
+        ]);
+    }
+
+    public function store(ProviderRequest $request): void
+    {
+        Provider::query()->create($request->validated());
+
+        $fileName = strtolower($request["name"]) . "." . $request->file("file")->getClientOriginalExtension();
+        $fileContents = $request->file("file")->get();
+
+        Storage::disk("public")->put("providers/" . $fileName, $fileContents);
+    }
+
+    public function update(ProviderRequest $request, Provider $provider): void
+    {
+        $provider->update($request->validated());
+
+        $imageName = strtolower($request["name"]) . "." . $request->file("file")->getClientOriginalExtension();
```

Comment:
Lest move that: `strtolower($request["name"]) . "." . $request->file("file")->getClientOriginalExtension();` to some common function in controller or helper for global use (and update other places with that).

Reply:
moved to getFileName()